### PR TITLE
chore(staff): Create sentry app and staff permission

### PR DESCRIPTION
**Pull request by schew2381**
Requires https://github.com/getsentry/getsentry/pull/13068

### CODE REVIEW

File: src/sentry/api/permissions.py [hunk 1]
```diff
@@ -75,11 +75,29 @@ class (that is not StaffPermission) require this mixin because staff does not gi
     https://www.python.org/download/releases/2.3/mro/ to learn more.
     """
 
-    def has_permission(self, request, *args, **kwargs):
-        return super().has_permission(request, *args, **kwargs) or is_active_staff(request)
```
[the remark this answers is not in the document]
Comment (schew2381) [reply src/sentry/api/permissions.py:79]: Hmm that's a good point. I'm trying to think it there's a reason to have the original check first

[the remark this answers is not in the document]
Comment (schew2381) [reply src/sentry/api/permissions.py:79]: I can't think of any reason unless someone stupidly puts a side effect in the permission check. Guess we'll find out if anything fails after I swap them

**Review by cathteng — APPROVED** [review]

File: src/sentry/api/permissions.py [hunk 2]
```diff
@@ -75,11 +75,13 @@ class (that is not StaffPermission) require this mixin because staff does not gi
...
+    def has_permission(self, request, *args, **kwargs) -> bool:
+        # Check for staff before calling super to avoid catching exceptions from super
+        return is_active_staff(request) or super().has_permission(request, *args, **kwargs)
```
Comment (ykamo001) [inline src/sentry/api/permissions.py:80]: we want the exceptions to bubble up?

Comment (schew2381) [reply src/sentry/api/permissions.py:80]: yes we catch them later in `dispatch` in api/base

Comment (ykamo001) [reply src/sentry/api/permissions.py:80]: but does it make sense to catch and log here, and simply return false?

Comment (schew2381) [reply src/sentry/api/permissions.py:80]: If we return `False` here the error msg ends up being a generic 403 and I think the raised exceptions sometimes have useful messages and codes

**Pull request merged**